src/main/java/net/datafaker/service/FakeValuesService.java

```diff
@@ -222,27 +219,7 @@ private String[] split(String string) {
         if (result != null) {
             return result;
         }
-        int size = 0;
-        final char splitChar = '.';
-        final int length = string.length();
-        for (int i = 0; i < length; i++) {
-            if (string.charAt(i) == splitChar) {
-                size++;
-            }
-        }
-        result = new String[size + 1];
-        final char[] chars = string.toCharArray();
-        int start = 0;
-        int j = 0;
-        for (int i = 0; i < length; i++) {
-            if (string.charAt(i) == splitChar) {
-                if (i - start > 0) {
-                    result[j++] = String.valueOf(chars, start, i - start);
-                }
-                start = i + 1;
-            }
-        }
-        result[j] = String.valueOf(chars, start, chars.length - start);
+        result = string.split("\\.");   // String#split is plenty efficient for single-char patterns
```

Comment:
This statement does not look like true

please provide proof for that.

Dummy test with jmh shows that current code is about 10-15% faster than `result = string.split("\\.");`

```
Benchmark             Mode  Cnt      Score      Error   Units
Issue760.testSplit   thrpt    5  24329.658 ±  734.929  ops/ms
Issue760.testSplit2  thrpt    5  21338.631 ± 1043.678  ops/ms
```
code snippet
```java
@BenchmarkMode(Mode.AverageTime)
@OutputTimeUnit(TimeUnit.MILLISECONDS)
@State(Scope.Benchmark)
@Fork(value = 2, jvmArgs = {"-Xms2G", "-Xmx2G"})
public class Issue760 {
    private static final String STRING_2_SPLIT = "Address.streetAddress";
    public static void main(String[] args) throws RunnerException {

        Options opt = new OptionsBuilder()
                .include(Issue760.class.getSimpleName())
                .forks(1)
                .build();

        new Runner(opt).run();
    }

    @Benchmark
    @BenchmarkMode(Mode.Throughput)
    public void testSplit(Blackhole blackhole) {
        final String string = STRING_2_SPLIT;
        int size = 0;
        final char splitChar = '.';
        final int length = string.length();
        for (int i = 0; i < length; i++) {
            if (string.charAt(i) == splitChar) {
                size++;
            }
        }
        String[] result = new String[size + 1];
        final char[] chars = string.toCharArray();
        int start = 0;
        int j = 0;
        for (int i = 0; i < length; i++) {
            if (string.charAt(i) == splitChar) {
                if (i - start > 0) {
                    result[j++] = String.valueOf(chars, start, i - start);
                }
                start = i + 1;
            }
        }
        result[j] = String.valueOf(chars, start, chars.length - start);
        blackhole.consume(result);
    }

    @Benchmark
    @BenchmarkMode(Mode.Throughput)
    public void testSplit2(Blackhole blackhole) {
        final String string = STRING_2_SPLIT;
        String[] result = string.split("\\.");
        blackhole.consume(result);
    }
}

```